src/librustc/infer/error_reporting.rs

```diff
@@ -523,7 +523,8 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
                          origin: TypeOrigin,
                          secondary_span: Option<(Span, String)>,
                          values: Option<ValuePairs<'tcx>>,
-                         terr: &TypeError<'tcx>)
+                         terr: &TypeError<'tcx>,
+                         implicit_return: bool)
```

Comment:
Could you rename this, and all other instances, to `is_block_tail`? It doesn't seem to be related to `return` at all.